packages/ui/src/components/table.tsx

```diff
@@ -181,7 +181,7 @@ function TableBody({
 
 Table.Dropdown = function TableDropdown({ children }: PropsWithChildren) {
   return (
-    <Dropdown className="fixed right-[unset] mt-[unset] translate-x-[calc(-100%-0.5rem)]">
+    <Dropdown className="fixed mr-4 translate-x-[calc(-0.5rem)]">
```

Comment:
So this is definitely better than the existing solution (and will be happy to merge), but not quite "the" solution that works in all cases.

This will only work when the table extends to a very specific of 16px to the left of the right edge of the page. That's because of the `right: 0` position, which is the default in `Dropdown`. The reason I'm okay with merging is b/c all tables right now extend to that `16px` point .

The idea I was going for before was unsetting the right position, which means that it'll automatically inherit the right position from its parent, but the reason it doesn't seem to be working is because when the table row is scrollable, the actual right position includes the scroll position, and thus it's pushed to the right of the screen.

If you can come up with a solution (ideally without Javascript) that works in a less hard-codey-ish way, that would be ideal but if not I'll merge it!

---

Also, nit but let's keep that `mt-[unset]`, I like the look of the dropdown being aligned to the top of the 3-dot button!

Reply:
Ah, I see your point. I've been playing around with different options, and when using `absolute right-[unset]` instead of `fixed right-[unset]`, the positioning is all great, except there's the big issue that when there are multiple dropdown elements, they are getting hidden and you have to scroll down to see them. 
<img width="449" alt="Screenshot 2024-05-09 at 9 08 39 AM" src="https://github.com/colorstackorg/oyster/assets/91388965/375c419d-536f-468c-9b69-48a38954952a">
<img width="452" alt="Screenshot 2024-05-09 at 9 09 10 AM" src="https://github.com/colorstackorg/oyster/assets/91388965/6ce5abf1-5575-4994-b133-4e26846ddb05">
This makes me think that if we want to confine the x-axis of the dropdown to auto-align with the table width (by using `absolute` instead of `fixed`, this forces the dropdown to also align with the vertical limits of the table. I'm not sure that there is a simple solution, since I think it would require separating `Table.Dropdown` from the rest of the table so that we can conform to the width of the parent, but not the vertical limits.

Reply:
Yeah I would love to use position `absolute` which is the default for our dropdown since it auto-aligns perfectly, but unfortunately the `overflow-scroll` on the table container is why the dropdown gets hidden underneath.

To recap our options:
1. Your initial solution, which fixes the issue for most of our use cases but is just a bit hard-codey.
2. We do position absolute, and introduce a different (but maybe a bit better?) usability bug of the dropdown being hidden underneath.
3. We do a fixed position, and then introduce some React/JS logic to set the `right` value. We'd have to take into account the screen size, scroll position, etc.

I'm genuinely torn b/c I'm not a huge fan of any of the options 😭 but 1 is probably my top choice considering the tradeoffs. What do you think?

I tried to find examples in the wild of tables that support 1) scrollable rows, 2) a sticky menu option on the right side of the row, and 3) dropdowns in the table, but I couldn't really find any. Maybe it's not the best design (I feel like it is though ugh)!

Reply:
In response to (2), the 'bug' won't always be noticeable because when a table has enough entries (which, in prod, would be most cases), the table size will be readjusted so no scrolling will be needed to see the dropdown elements. However this is still essentially introducing a bug so that doesn't seem like a good option. 

(3) seems like it could be unnecessarily complicated since there'd be a lot of positioning considerations that could potentially introduce different bugs when used on different devices/screens, etc., and this could end up making it more complicated to make other tables in the future. 

I'm leaning towards (1) as well, because even though it is a bit hard-codey, it does fit with all the uses of tables now, and I wonder if in the future a different kind of table is needed, maybe it'd be worth creating a new table dropdown element if needed, and leave the current one as the main dropdown.

Reply:
Okay let's go with option 1!